onboard/inventory_entry_iterator.go

```diff
@@ -0,0 +1,69 @@
+package onboard
+
+import (
+	"errors"
+
+	"github.com/treeverse/lakefs/catalog/rocks"
+	"github.com/treeverse/lakefs/cmdutils"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type inventoryEntryIterator struct {
+	it       *InventoryIterator
+	value    *rocks.EntryRecord
+	err      error
+	progress *cmdutils.Progress
+}
+
+func NewValueToEntryIterator(it *InventoryIterator, progress *cmdutils.Progress) rocks.EntryIterator {
+	return &inventoryEntryIterator{
+		it:       it,
+		progress: progress,
+	}
+}
+
+func (e *inventoryEntryIterator) Next() bool {
+	if e.err != nil {
+		return false
+	}
+	if !e.it.Next() {
+		e.value = nil
+		e.err = e.it.Err()
+		return false
+	}
+	v := e.it.Get()
+
+	e.value = &rocks.EntryRecord{
+		Path: rocks.Path(v.Obj.Key),
+		Entry: &rocks.Entry{
+			Address: v.Obj.PhysicalAddress,
+			Size:    v.Obj.Size,
+			ETag:    []byte(v.Obj.Checksum),
```

Comment:
Think the original idea was to keep it as bytes - so in case the checksum is in hex we need to encode

Reply:
Done.

Reply:
1. You can remove the assignment of ETag now from line 48
2. Just make sure that import checksum is actually in Hex format and not base64.